### ISSUE TITLE
Gibberish (random?) output after I say only "hello".

### ISSUE THREAD

**Issue by jaggzh** [LAION-AI/natural_voice_assistant#6]
The assistant begins saying:
```
\`\`\`def count_vowels(word):    vowels = ['a', 'e', 'i', 'o', 'u']    count = 0    for letter in word
## Total Latency:  0.545
:        if letter.
```
... and she continues. :)

I thought it was the result of local echo, so I enabled module-echo-cancel (and tested it and it completely inhibits feedback). Unfortunately it has no effect on the assistant outputting continual gibberish.) :)

**Comment by braunale** [comment]
Hey, thanks for your message! Did you checked, if your speech was transcribed correclty? In the console it should print out the transcribed user query, so you can check what exactly is fed into the LLM. Could you try it again, so we can check if the problem occurs on the text generation or already on the transcription step.

**Comment by braunale** [comment]
I looked into this topic further. We are currently using phi-2 as our language model. Regarding to the model description (https://huggingface.co/microsoft/phi-2), you can use two different ways to format prompts either using Instruct:<prompt>Output: for instructions and fact-questions or Alice:<prompt>Bob: for conversations. Currently we are using the Instruct/Output pattern which works well for questions like "How many people live in france?" but breaks at smalltalk inputs. This is most likely the cause for the behavior you are describing. 

At the moment we unfortunately can't do very much against it, since this problem is caused by the LLM itself. We could of course use a larger LLM which will be more robust but for the cost of more latency. As statetd in our roadmap, we are trying to reduce the latency as much as possible and are especially aiming for a fully decoder-only architecture that would allow to stream a single token trhough STT, LLM and TTS. With this, we would be able to use much larger models and still ensure real-time responses. With a larger LLM the outputs will be much more stable. 

So to sum it up: The problem you are describing is most likely not caused by our code but by the small LLM we are using at the moment. Solving this issue is part of our general roadmap. So I will close this issue for now. If you have any ideas or suggestions for intermediate solutions that could improve the quality of the LLM right now or want to helpt to reduce the latency of the pipeline in order to be able to use larger LLMs, you are very welcome to open another issue or a PullRequest. Thanks for your collaboration!
